Improve separation/alignment of Editor Node panel

fregante opened the pull request:
## What does this PR do?

- Mentioned in https://pixiebrix.slack.com/archives/C03CA2M4RCM/p1707565424228409

In short, the sizing and alignment of the labels in this component, seem off because there's no visual separation between this component and the following fields:

<img width="886" alt="Screenshot 13" src="https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/973371f6-adcb-4869-a1b2-874103ed28ca">




## Demo

<details>


<summary>oudated demo</summary>



I noticed that Mods use the Card style:

<img width="886" alt="Screenshot 15" src="https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/a0f281c9-7f12-4b96-ac68-2d1c88438952">

So I thought I'd reuse it here:


<img width="886" alt="Screenshot 14" src="https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/818f4468-c4b5-451e-b696-fdf02fd6ed0f">



</details>

## Discussion

- I tried using the same style as `CollapsibleFieldSection`, but it's impossible because some fields _already_ wrap their own options in multiple sections, while simple ones don't.

## Review tips

- "Hide whitespace changes" 

## Checklist

- [x] Designate a primary reviewer: @grahamlangford

fregante commented:
Here's a wider view for reference:

<img width="1016" alt="Screenshot" src="https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/859bf7cc-266a-470e-8f41-4ee088b86811">

fregante commented on src/pageEditor/fields/CollapsibleFieldSection.tsx:
```diff
@@ -43,14 +43,12 @@ const CollapsibleFieldSection: React.FC<{
     <div className={styles.root}>
-      <div
-        role="button"
-        tabIndex={0}
```
Real buttons are always preferable

twschiller commented:
cc @BrandonPxBx In general, I think we've been wanting to move away from card-heavy layouts (especially nested cards). @BrandonPxBx do you have a perspective? This is coming up because on MV3 there's less width in the Page Editor

fregante commented:
And this is what it would look like with the aforementioned "collapsible sections inside BrickOptions"

<img width="1004" alt="Screenshot" src="https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/8a391aa5-f834-406d-a7da-e81ffc4f829a">

grahamlangford commented:
I also want to call attention to this bit of misalignment:
<img width="134" alt="image" src="https://github.com/pixiebrix/pixiebrix-extension/assets/30706330/c2a7b880-6151-42ca-ac97-96bce83988c9">

BrandonPxBx commented:
No approval on the use of the card. A simple divider or some more whitespace between the two inputs could help. 

But the ideal scenario is to correct the spacing. No point in making a change just to make a change if it's not going to fix the problem.

fregante commented:
> I also want to call attention to this bit of misalignment

Yes that's always been there. We have literally no choice here because the wrapping happens "when it doesn't fit" so we can't control the the width of the labels at that point. 

CSS grid might help here, and I'll try it, but I'm not confident about it due to the DOM structure. 

The simplest solution is to not place two fields on a single line.

> No approval on the use of the card

We agree here

> No point

The point is do add separation, which is valid regardless of alignment. Unfortunately due to the flexibility of the fields below, we just cannot have a line or a header. See all the screenshots above.

fregante commented:
Ok, I fixed the issue when the fields in a column:

<img width="984" alt="Screenshot" src="https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/16200d33-c9a7-45cc-9801-251e499ba975">

But that cannot be fixed when they're in a row, so I added a background color to slightly box them:

<img width="984" alt="Screenshot 1" src="https://github.com/pixiebrix/pixiebrix-extension/assets/1402241/efc1ceff-8667-4ae1-ba7f-677eb664d5a0">

If you know of a better solution, please mockup

BrandonPxBx commented:
> The point is do add separation, which is valid regardless of alignment. Unfortunately due to the flexibility of the fields below, we just cannot have a line or a header. See all the screenshots above.

If the correct fix is too difficult, then no fix should be implemented. I do not want to introduce a new design pattern for the sake of a new design pattern with little to no user feedback that this is currently a problem

BrandonPxBx reviewed (changes requested):
I do not approve this change. There's no reason to make this change.

fregante commented:
Bugfix merged, closing this PR
